### ISSUE TITLE
Allow Service Endpoints Use

### ISSUE THREAD

**Issue by calebcartwright** [swellaby/vsts-mirror-git-repository#1]
We should allow the task configuration to point to existing service endpoints for specifying the git repo and authentication info (for example the GitHub and Bitbucket service endpoints).

This would provide an option for folks to use existing connections.

**Comment by traviskosarek** [comment]
I agree. I think this would be a good thing to add. It makes a lot sense.

In terms of endpoints, I think we can support some of the already existing service endpoints (I'm thinking Github specifically), but should we also include a Generic Git service endpoint with the task?

And if we do that, would that completely replace the "in task" repository endpoint, or would that just be another option?

I know we can do it either way, but I want to hear other opinions before I go do it ;)

**Comment by calebcartwright** [comment]
it may not make sense to do them all at once (can add support incrementally), but I would think we should support all of them. 

All VSTS accounts should automatically and already having an existing Service Endpoint types for [GitHub](https://docs.microsoft.com/en-us/vsts/build-release/concepts/library/service-endpoints#sep-github), [Bitbucket](https://docs.microsoft.com/en-us/vsts/build-release/concepts/library/service-endpoints#sep-bbucket), as well as a [generic Git endpoint type](https://docs.microsoft.com/en-us/vsts/build-release/concepts/library/service-endpoints#sep-extgit), so I don't think we'll have to make a custom endpoint type.
